src/main/java/io/vertx/ext/mail/impl/sasl/ByteArrayBuilder.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+package io.vertx.ext.mail.impl.sasl;
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.CoderResult;
+import java.nio.charset.CodingErrorAction;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Builder class for sequences of bytes.
+ *
+ * This copies from https://github.com/apache/httpcomponents-client/blob/5.0.3-RC1/httpclient5/src/main/java/org/apache/hc/client5/http/utils/ByteArrayBuilder.java
+ */
+class ByteArrayBuilder {
```

Comment:
Why not just use `io.vertx.core.Buffer` ?

Reply:
I think we can do the same with our own classs instead of this copy of http-commons. But I didn't look for all the details, so my view is limited.

Reply:
@pmlopes  thanks for the review :) it makes sense to use `io.vertx.core.Buffer` as the only use of `ByteArrayBuilder` is to convert the `char[]` to `byte[]`. Updated accordingly.